module/solutions/helper.js

```diff
@@ -142,4 +1012,2537 @@ module.exports = class SolutionsHelper {
+          return resolve({
+            message: CONSTANTS.apiResponses.INVALID_LINK,
+            result: [],
+          });
+        }
+
+        if (solutionData[0].status !== CONSTANTS.common.ACTIVE_STATUS) {
+          return resolve({
+            message: CONSTANTS.apiResponses.LINK_IS_EXPIRED,
+            result: [],
+          });
+        }
+
+        if (
+          solutionData[0].endDate &&
+          new Date() > new Date(solutionData[0].endDate)
+        ) {
+          if (solutionData[0].status === CONSTANTS.common.ACTIVE_STATUS) {
+            let updateSolution = await this.update(
+              solutionData[0]._id,
+              {
+                status: CONSTANTS.common.INACTIVE,
+              },
+              userId
+            );
+          }
+
+          return resolve({
+            message: CONSTANTS.apiResponses.LINK_IS_EXPIRED,
+            result: [],
+          });
+        }
+
+        response.verified = true;
+        return resolve({
+          message: CONSTANTS.apiResponses.LINK_VERIFIED,
+          result: response,
+        });
+      } catch (error) {
+        return resolve({
+          success: false,
+          status: error.status
+            ? error.status
+            : HTTP_STATUS_CODE.internal_server_error.status,
+          message: error.message,
+        });
+      }
+    });
+  }
+
+
+
+    /**
+   * Add entities in solution.
+   * @method
+   * @name addEntitiesInScope
+   * @param {String} solutionId - solution Id.
+   * @param {Array} entities - entities data.
+   * @returns {JSON} - Added entities data.
+   */
+
+    static addEntitiesInScope(solutionId, entities) {
+      return new Promise(async (resolve, reject) => {
+        try {
+          let solutionData = await solutionsQueries.solutionsDocument(
+            {
+              _id: solutionId,
+              scope: { $exists: true },
+              isReusable: false,
+              isDeleted: false,
+            },
+            ["_id", "programId", "scope.entityType"]
+          );
+  
+          if (!solutionData.length > 0) {
+            return resolve({
+              status: HTTP_STATUS_CODE.bad_request.status,
+              message: CONSTANTS.apiResponses.SOLUTION_NOT_FOUND,
+            });
+          }
+          let programData = await programQueries.programsDocument(
+            {
+              _id: solutionData[0].programId,
+            },
+            ["scope.entities", "scope.entityType"]
+          );
+  
+          if (!programData.length > 0) {
+            return resolve({
+              status: HTTP_STATUS_CODE.bad_request.status,
+              message: CONSTANTS.apiResponses.PROGRAM_NOT_FOUND,
+            });
+          }
+  
+          if (
+            solutionData[0].scope.entityType !== programData[0].scope.entityType
+          ) {
+            let matchData = [];
+            let checkEntityInParent = [];
+            let childEntities = await userService.getSubEntitiesBasedOnEntityType(
```

Comment:
@Prajwal17Tunerlabs it is entity service

Reply:
removed getSubEntitiesBasedOnEntityType dependency